posts/2024-12-13_document_yo.../document_your_universe_with_dverse.qmd

```diff
@@ -34,13 +34,46 @@ The example here uses the 'admiralverse' -- a toy universe defined as every
 package on CRAN today (2024-12-07), with a website, and a name starting with
 'admiral'.
 
+```{r admiralverse}
+admiralverse <- c(
+  "admiral",
+  "admiraldev",
+  "admiralonco",
+  "admiralophtha",
+  "admiralpeds",
+  "admiralvaccine"
+)
+```
+
 You need:
 
 * the package {dverse} to <u>d</u>ocument your uni<u>verse</u>, 
 * each package in the toy 'admiralverse' to access the documentation metadata,
 * {dplyr} to manipulate the output of {dverse}, and 
 * {DT} to create a searchable table.
 
+```{r dependencies}
+#| echo: false
+#| message: false
+
+# Ensure the packages used here are installed; else install them in a local
+# library to avoid forcing new dependencies in the system of the blog
+# administrators
+
+local_library <- tempfile()
+dir.create(local_library)
+# https://adv-r.hadley.nz/functions.html#on-exit
+on.exit(unlink(local_library), add = TRUE, after = FALSE)
+
+.libPaths(c(.libPaths(), local_library))
+
+if (!requireNamespace("pak", quietly = TRUE)) {
+  install.packages("pak", lib = local_library)
+}
+# The cache is smart. It won't re-install what's already available
+pak::pak(c("dverse", admiralverse, "dplyr", "DT"), lib = local_library)
+```
```

Comment:
Here I ensure that the packages used here are installed; else I install them in a local library to be polite -- I want to avoid forcing new dependencies in your system.

Let me know if you do prefer me to install them at a more general level, maybe in the Dockerfile?

Reply:
We install them as part of a workflow - I have added them now and so can close this PR :) blog is live!